src/soc/ibm/power9/chip.c

```diff
@@ -745,10 +749,43 @@ static void activate_slave_cores(uint8_t chip)
 	}
 }
 
+static void * load_fdt(const char *dtb_file, uint8_t chips)
+{
+	void *fdt;
+	void *fdt_rom;
+	struct device_tree *tree;
+
+	fdt_rom = cbfs_map(dtb_file, NULL);
+	if (fdt_rom == NULL)
+		die("Unable to load %s from CBFS\n", dtb_file);
+
+	tree = fdt_unflatten(fdt_rom);
+
+	dt_platform_update(tree, chips);
+
+	fdt = malloc(dt_flat_size(tree));
+	if (fdt == NULL)
+		die("Unable to allocate memory for flat device tree\n");
+
+	dt_flatten(tree, fdt);
+	return fdt;
+}
+
 void platform_prog_run(struct prog *prog)
 {
 	uint8_t chips = fsi_get_present_chips();
 
+	void *fdt;
+	const char *dtb_file;
+
+	assert(chips == 0x01 || chips == 0x03);
+
+	dtb_file = (chips == 0x01 ? "1-cpu.dtb" : "2-cpus.dtb");
+	fdt = load_fdt(dtb_file, chips);
+
+	/* See asm/head.S in skiboot where fdt_entry starts at offset 0x10 */
+	prog_set_entry(prog, (void *)0x10, fdt);
```

Comment:
In case payload is loaded to a different address:
```suggestion
	prog_set_entry(prog, prog_start(prog) + 0x10, fdt);
```